### PR TITLE
#38  Execute infrastructure automated tests locally

### PR DESCRIPTION
**Pull request by code4clouds**
Code for Issue #38.  It includes the assert code for azure and sample files for the users.

### CODE REVIEW

File: tests/assert.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    export rg=$1
+    export resource_type=$2  # i.e. Microsoft.KeyVault/vaults
+    export resource_name=$3   
+
+    printf "\e[1;37mTesting Resource Group: $rg, Type: $resource_type, Name: $resource_name - "
+
+    if [ ${4:-"yes"}  == "not" ] # Set the assert type.  If assert_type is empty check for exist, else validate for not exist
+    then
+    echo "not exist"
+    az resource list -g $rg --resource-type $resource_type | jq  -e 'if contains( [{ name: env.resource_name }] ) then false else true end' > /dev/null
```
Comment (rickrain) [inline tests/assert.sh]: The dependency on jq just doesn't seem necessary.  For example, the built-in Azure CLI query commands could achieve the same results.  Here's an example:
```
resourceId=""
resourceId=(az resource show --resource-group $rg --resource-type $resource_type --name $resource_name --query 'id')
if [[ -z $resourceId ]]; then
  # resource exists
else
  # resource does not exist
fi
```

Comment (code4clouds) [reply tests/assert.sh]: fixed in new commit... no longer jq is required

File: tests/test.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+assertRG expected-resource-group-name
```
[the remark this answers is not in the document]
Comment (code4clouds) [reply tests/test.sh]: This test operation is only for check the creation of a resource group only.   The other test operation is for checking of resources especifically.

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+## Requirements
+
+- Azure CLI
+- jq (apt-get install jq)
+
+## Description
+
+- assert.sh is the code used for the test functions needed for the test files.
+- main.sh is the file that initiates the tests
+- test.sh is an example file for testing the existence of resource groups in a subscription
```
[the remark this answers is not in the document]
Comment (code4clouds) [reply tests/README.md]: The resource test operation is only for check the creation of a resource group only. The other test operation is for checking of resources especifically.

File: tests/assert.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+function printResult {
+    if [ $1 -eq 1 ]
+        then printf "\e[0;31mFailed...\n"
+        else printf "\e[0;32mSuccess...\n"
+    fi
+}
+
+# Search a resource group inside Azure"
+function assertRG {
+    export name=$1
```
[the remark this answers is not in the document]
Comment (code4clouds) [reply tests/assert.sh]: fixed in new commit... no longer jq is required

File: tests/assert.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+if [ $1 == "continue_on_error" ]
+then
+    set +e 
+else  # stop on error
+    set -e
+fi
+
+function printResult {
```
[the remark this answers is not in the document]
Comment (code4clouds) [reply tests/assert.sh]: fixed in new version.

File: tests/assert.sh [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    if [ $1 -eq 1 ]
+        then printf "\e[0;31mFailed...\n"
+        else printf "\e[0;32mSuccess...\n"
+    fi
+}
+
+# Search a resource group inside Azure"
+function assertRG {
+    export name=$1 
+    printf "\e[1;37mTesting Resourgroup $name - "
```
[the remark this answers is not in the document]
Comment (code4clouds) [reply tests/assert.sh]: no longer apply in new commit

File: tests/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+- test2.sh is an example file for testing the existence resources in a resource group
+ 
+## Asserts
+
+The followng command chech for the existence of these resource (or resource gruop)
```
[the remark this answers is not in the document]
Comment (code4clouds) [reply tests/README.md]: fixed

File: tests/README.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+- Azure CLI
+- jq (apt-get install jq)
+
+## Description
+
+- assert.sh is the code used for the test functions needed for the test files.
+- main.sh is the file that initiates the tests
+- test.sh is an example file for testing the existence of resource groups in a subscription
+- test2.sh is an example file for testing the existence resources in a resource group
```
[the remark this answers is not in the document]
Comment (code4clouds) [reply tests/README.md]: what are your thoughts on names for the sample tests?

[the remark this answers is not in the document]
Comment (code4clouds) [reply tests/README.md]: renamed the files as sampleTest1 & 2

**Comment by rickrain** [comment]
Closing PR and merged into master.

**Pull request merged**